numba/np/ufunc_db.py

```diff
@@ -620,6 +620,10 @@ def _fill_ufunc_db(ufunc_db):
         'FF->?': npyfuncs.np_complex_gt_impl,
         'DD->?': npyfuncs.np_complex_gt_impl,
     }
+    if numpy_version >= (1, 25):
```

Comment:
I'm not sure this is happening - there don't appear to be any tests generated that involve `qQ` or `Qq`:

```
$ python runtests.py numba.tests.test_ufuncs -l | egrep "(qQ|Qq)"
$
```

(i.e. no tests with signed / unsigned are generated).

Reply:
Additionally, `BasicUFuncTest.basic_ufunc_test()` isn't set up to test binary functions with inputs of different types - it is set up to assume arguments are homogeneous in type:

https://github.com/numba/numba/blob/766f6d9b250ebf452c75590f753de0c003563541/numba/tests/test_ufuncs.py#L176-L184

(there is only one `input_type`)

Reply:
My "egrep" check earlier was run erroneously because I didn't have NumPy 1.25 in my environment. The `TestLoopTypes` test classes were generated `qQ` and `Qq` variants of the tests. However, the `BasicUFuncTest` class didn't handle these combinations of types, so it was not testing on CUDA.

I tried experimenting with adding testing of these types to the `BasicUFuncTest` class as well - the commit https://github.com/gmarkall/numba/commit/865cfc6aea5aec24d97a0c5e961c4ac370a13580 on the branch https://github.com/gmarkall/numba/tree/numpy_1.25 implements this (and therefore tests these implementations with CUDA too) - feel free to pick it up / modify it if you'd like.

Reply:
Cherry-picked.